src/cluster.c

```diff
@@ -7005,15 +7005,25 @@ void slotToKeyDestroy(redisDb *db) {
  * The number of removed items is returned. */
 unsigned int delKeysInSlot(unsigned int hashslot) {
     unsigned int j = 0;
+    server.propagate_no_multi = 1;
     dictEntry *de = (*server.db->slots_to_keys).by_slot[hashslot].head;
     while (de != NULL) {
         sds sdskey = dictGetKey(de);
         de = dictEntryNextInSlot(de);
         robj *key = createStringObject(sdskey, sdslen(sdskey));
         dbDelete(&server.db[0], key);
+        signalModifiedKey(NULL, &server.db[0], key);
+        moduleNotifyKeyspaceEvent(NOTIFY_GENERIC, "del", key, server.db[0].id);
+        propagateDeletion(&server.db[0], key, server.lazyfree_lazy_server_del);
         decrRefCount(key);
         j++;
+        server.dirty++;
     }
+
+    /* Propagate all DELs */
+    propagatePendingCommands();
+    server.propagate_no_multi = 0;
```

Comment:
now that #10969 got merged.
we need to move propagatePendingCommands into the loop, and drop the use of propagate_no_multi (doesn't exist anymore)